app/uk/gov/hmrc/selfservicetimetopay/models/EligibilityStatus.scala

```diff
@@ -26,7 +26,7 @@ case object DebtIsInsignificant extends Reason("DebtIsInsignificant")
 case object OldDebtIsTooHigh extends Reason("OldDebtIsTooHigh")
 case object TotalDebtIsTooHigh extends Reason("TotalDebtIsTooHigh")
 case object ReturnNeedsSubmitting extends Reason("ReturnNeedsSubmitting")
-case object DirectDebitCreatedWithinTheLastYear extends Reason("DirectDebitCreatedWithinTheLastYear")
+case object DirectDebitCreatedWithinTheLastTwoYears extends Reason("DirectDebitCreatedWithinTheLastTwoYears")
```

Comment:
is this gonna affect existing data in mongo that has the old case object name written in it?

Reply:
that's a fair point, would that be an issue?

Reply:
if this is written in the user's session data then if this new code is deployed then any existing sessions with `DirectDebitCreatedWithinTheLastYear` would break cos the service wouldn't be able to read that value anymore. That may not be very likely to happen and would only be a temporary thing so it might not be the end of the world. Might want to double check if the `EligibilityStatus` is written anywhere else though just in case to see if there'd be any additional impact.

If we're renaming stuff then having a more general name like `DirectDebitAlreadyCreated` would be better in case we change the rules again in the future.

Reply:
Will it? It will just write a different string as a reason in the 
"maybeEligibilityStatus" : {
        "reasons" : [
"DirectDebitCreatedWithinTheLastYear"
        ]
    }

Reply:
Should tell Adam, because he may be tracking the reason for auditing.

Reply:
it appears the reads instance for `Reason` uses this

https://github.com/hmrc/self-service-time-to-pay-frontend/blob/75a9c7b6139812bab63a9f11d9044561dffa4bd1/app/uk/gov/hmrc/selfservicetimetopay/models/package.scala#L48

which looks like this:
```scala
  def parseFromString(jsonString: String): Option[Reason] = jsonString.trim.toLowerCase match {
    case "nodebt"                                 => Some(NoDebt)
    case "debtisinsignificant"                    => Some(DebtIsInsignificant)
    case "olddebtistoohigh"                       => Some(OldDebtIsTooHigh)
    case "totaldebtistoohigh"                     => Some(TotalDebtIsTooHigh)
    case "ttpislessthentwomonths"                 => Some(TotalDebtIsTooHigh)
    case "directdebitcreatedwithinthelastyear"    => Some(DirectDebitCreatedWithinTheLastYear)
    case x if x.contains("returnneedssubmitting") => Some(ReturnNeedsSubmitting)
    case _ =>
      None
  }
```
Not sure what's going on there though cos not all the reasons have a mapping and does it really match up with what the writes instance is doing? 🤔 The writes instance is `toString`-ing the reason which doesn't look like would give what this pattern match is looking for. Are we actually writing these reasons to mongo?

Reply:
`toString`-ing the case objects and then `toLowerCase` would give the things in the pattern match I guess - but then why aren't all the reasons accounted for in the pattern match? 🤔

Reply:
the name makes sense for sure, I'll change it to DirectDebitAlreadyCreated; is there a way I can take a look at the session data that's created?

Reply:
if you've got something like studio3T then you can inspect what's in mongo through there - otherwise you can play around in the tests and maybe even think about adding tests to make sure the serialisation and deserialisation of these reason work properly

Reply:
I do have studio 3t; I dont see any of this in the ssttp mongo

Reply:
are you sure? I think I can see it on my machine